### DISCUSSION TITLE
LCD demo script

### DISCUSSION THREAD

**Discussion by Richrfl** [FrSkyRC/ETHOS-Feedback-Community#2959] [Q&A] [answered]
I am trying to understand the parameters to build up a widget with 5 or six lines of text, based on the Lua LCD demo widget. The problem is that I cannot figure out what parameters I have to change to be able to write more than three lines. I tried changing the line "lcd.drawBitmap" using "bipmap, w2, h2" but that didn't help.
Here is the code I am using. I want to add commented lines.

```cpp
-- Lua LCD demo widget

local bitmap, mask
local function paint(widget)
    -- Demo of a mask
--    lcd.color(lcd.RGB(255, 255, 0))
--    lcd.drawMask(0, 30, mask)
    -- Demo of a text
    lcd.font(FONT_ITALIC)
    lcd.font(FONT_L)
    lcd.color(lcd.RGB(255,255,255))
    lcd.drawText(15, 0, "FS1: Main lights switch")
	lcd.drawText(15, 20, "SD DOWN and SI screenshot")
	lcd.drawText(15,40,"SG Reverse Motor Switch")
--	lcd.drawText(15,60,"FS1 Nav lights")
--	lcd.drawText(15,60,"FS2 Reverse Lights Switch)
--	lcd.drawText(15,100,"SJ and RS>95%: RST Log")
    -- Demo of a bitmap
    local w2, h2 = lcd.getWindowSize()
    lcd.drawBitmap(10, 25, bitmap, w2 - 20, h2 - 35)
end
local function init()
    bitmap = lcd.loadBitmap("radio/bitmaps/models/default_plane.png")
    mask = lcd.loadMask("FLASH:/bitmaps/system/mask_dot.png")
    system.registerWidget({key="lcddemo", name="PC_21", paint=paint})
end
return {init=init}
```

**Comment by bsongis-frsky** [comment, marked as answer]
You miss some double quotes on this line, no?
`lcd.drawText(15,60,"FS2 Reverse Lights Switch)`

**Reply by Richrfl** [reply]
It seems that that was the whole problem. Now I am able to write at least 6 lines.
I have another issue on the Sim, and you tell me if I publish it somewhere else: In the model, I select the PC-21 script (which is the one we are looking up now), but if I close and re-open the Sim, the model shows a DH-6 script. Kind is not saving my wiglet setting.

**Reply by bsongis-frsky** [reply]
Perhaps the two scripts have the same scriptId? It should be unique.

**Reply by Richrfl** [reply]
Where it is the scrptid?
For example in the code I sent?
